react/src/views/Transfers/TransferAgreementOverview/TransferAgreementOverview.test.tsx

```diff
@@ -0,0 +1,286 @@
+      mockSuccessfullTransferAgreementsQuery({
+        type: TransferAgreementType.SendingTo,
+        isInitiator: false,
+      }),
+      mockGraphQLError(REJECT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Reject/i,
+    toastText: /could not reject/i,
+  },
+  {
+    name: "4.2.4.1b - Reject Transfer Agreement fails due to NetworkError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({
+        type: TransferAgreementType.ReceivingFrom,
+        isInitiator: false,
+      }),
+      mockNetworkError(REJECT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Reject/i,
+    toastText: /could not reject/i,
+  },
+  {
+    name: "4.2.5.1a - Cancel Transfer Agreement fails due to GraphQLError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ state: TransferAgreementState.Accepted }),
+      mockGraphQLError(CANCEL_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /accepted/i,
+    modalButtonText: /terminate/i,
+    toastText: /could not cancel/i,
+  },
+  {
+    name: "4.2.5.1b - Cancel Transfer Agreement fails due to NetworkError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ state: TransferAgreementState.Accepted }),
+      mockNetworkError(CANCEL_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /accepted/i,
+    modalButtonText: /terminate/i,
+    toastText: /could not cancel/i,
+  },
+];
+
+failedMutationTests.forEach(({ name, mocks, stateButtonText, modalButtonText, toastText }) => {
+  it(name, async () => {
+    const user = userEvent.setup();
+    render(<TransferAgreementOverviewView />, {
+      routePath: "/bases/:baseId/transfers/agreements",
+      initialUrl: "/bases/1/transfers/agreements",
+      mocks,
+    });
+
+    // click the button in the state column
+    const stateButton = await screen.findByRole("button", { name: stateButtonText });
+    expect(stateButton).toBeInTheDocument();
+    user.click(stateButton);
+
+    // click the button in the modal
+    const modalButton = await screen.findByRole("button", { name: modalButtonText });
+    expect(modalButton).toBeInTheDocument();
+    user.click(modalButton);
+
+    // error toast shown and overlay is still open
+    expect(await screen.findByText(toastText)).toBeInTheDocument();
+    expect(modalButton).toBeInTheDocument();
+  });
+});
+
+const mockSuccessfullMutation = ({
+  mutation = ACCEPT_TRANSFER_AGREEMENT,
+  mutationKey = "acceptTransferAgreement",
+  state = TransferAgreementState.UnderReview,
+  type = TransferAgreementType.Bidirectional,
+  isInitiator = true,
+}) => {
+  const mockObject = {
+    request: {
+      query: mutation,
+      variables: { id: "1" },
+    },
+    result: {
+      data: {},
+    },
+  };
+  mockObject.result.data[mutationKey] = generateMockTransferAgreement({ state, type, isInitiator });
+  return mockObject;
+};
+
+const succesfullMutationTests = [
```

Comment:
here and below: `successful` instead of `succesfull` or `successfull`

Reply:
done